src/main/scala/mesosphere/marathon/core/appinfo/impl/AppInfoBaseData.scala

```diff
@@ -173,6 +173,37 @@ class AppInfoBaseData(
     val statusSince = if (instanceStatus.isEmpty) now else instanceStatus.map(_.statusSince).max
     val state = await(podState(podDef.instances, instanceStatus, isPodTerminating(podDef.id)))
 
+    val taskFailureOpt = await {
+      taskFailureRepository
+        .get(podDef.id)
+        .recover { case NonFatal(e) => None }
+    }
+    val failedInstanceBundle = taskFailureOpt.flatMap { taskFailure =>
+      val failedTaskId = core.task.Task.Id(taskFailure.taskId)
+      instances.collectFirst {
+        case instance if instance.tasksMap.keySet.contains(failedTaskId) =>
```

Comment:
Why not `instance.tasksMap.contains(failedTaskId)`? `.keySet` will do another copy if I'm not mistaken.